diff --git a/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgView.java b/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgView.java
index 82dc328..fc5e60a 100644
--- a/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgView.java
+++ b/node_modules/react-native-svg/android/src/main/java/com/horcrux/svg/SvgView.java
@@ -67,6 +67,7 @@ public class SvgView extends ReactViewGroup implements ReactCompoundView, ReactC
     }
 
     private @Nullable Bitmap mBitmap;
+    private boolean mRemovalTransitionStarted = false;
 
     public SvgView(ReactContext reactContext) {
         super(reactContext);
@@ -88,13 +89,33 @@ public class SvgView extends ReactViewGroup implements ReactCompoundView, ReactC
                 return;
             }
             mRendered = false;
-            ((VirtualView) parent).getSvgView().invalidate();
+            SvgView svgView = ((VirtualView) parent).getSvgView();
+            if(svgView != null) {
+                svgView.invalidate();
+            }
             return;
         }
-        if (mBitmap != null) {
-            mBitmap.recycle();
+        
+        if(!mRemovalTransitionStarted){
+            if (mBitmap != null) {
+                mBitmap.recycle();
+            }
+            mBitmap = null;
+        }
+    }
+            
+    @Override
+    public void startViewTransition(View view) {
+        super.startViewTransition(view);
+                    mRemovalTransitionStarted = true;
+    }
+            
+    @Override
+    public void endViewTransition(View view) {
+        super.endViewTransition(view);
+        if (mRemovalTransitionStarted) {
+            mRemovalTransitionStarted = false;
         }
-        mBitmap = null;
     }
 
     @Override
